feature/gribi/ate_tests/ipv4_entry_test/README.md

```diff
@@ -0,0 +1,65 @@
+# TE-2.1: gRIBI IPv4 Entry
+
+## Summary
+
+Validate IPv4 support in gRIBI.
+
+## Procedure
+
+*   Connect ATE port-1 to DUT port-1, ATE port-2 to DUT port-2, and ATE port-3
+    to DUT port-3.
+*   Establish gRIBI client connection with DUT negotiating RIB_AND_FIB_ACK as
+    the requested ack_type.
+*   Using gRIBI Modify RPC install the following IPv4Entry sets, and validate
+    the specified behaviours:
+    *   Single IPv4Entry -> NHG -> NH.
+        *   Install 198.51.100.0/24 to NextHopGroup containing one NextHop
+            specified to ATE port-2.
+        *   Forward packets between ATE port-1 and ATE port-2 (destined to
+            198.51.100.0/24 ) and determine that packets are forwarded
+            successfully:
+    *   Single IPv4Entry -> NHG -> multiple NHs.
+        *   Install 198.51.100.0/24 to NextHopGroup containing two NextHop
+            entries specified to ATE ports 2 and 3.
+        *   Validate that packets forwarded between ATE ports 1 and (2 and 3),
+            ensuring that traffic is forwarded.
+    *   Single IPv4Entry -> NHG -> non-existent NH.
+        *   Send a Modify() containing 2 AFTOperations that Install
+            198.51.100.0/24 to NextHopGroup containing next-hops that do not
+            exist. Validate that FAILED error is received for all the 2
+            operations. Ensure that traffic to 198.51.100.0/24 is blackholed.
+    *   Single IPv4Entry -> NHG -> NH with down interface
```

Comment:
(Since you are already writing it, I'm making suggestions here instead of the test plan). 

Can you help update it to: 
```
Install 198.51.100.0/24 to NextHopGroup containing a NextHop that references (interface_ref) a down interface and override the destination MAC (mac_address), ensure that FIB_PROGRAMMED is returned.
```
This will be more aligned with our prod goal.

@nflath to help LGTM my proposal.

Reply:
Could you elaborate a little more on "and override the destination MAC (mac_address)"?

* Is the MAC address override being specified for the NextHop using GRIBI, or by OC interacting with FDB?
* Whose DMAC is being used?
* Is traffic now expected to work?  This case as originally written is expected for traffic to fail.

Reply:
* `mac_address` and `interface_ref` (both specified by gRIBI NextHop) comes together in our use case.
* We can use `00:1A:11:00:00:01`.
* `FIB_PROGRAMMED` but does not expect the traffic to work.

Reply:
cleared @bstoll 's question in a offline chat. waiting for @nflath to LGTM.

Reply:
LGTM